### ISSUE TITLE
Error decoding c1#pad1.tmp.cram

### ISSUE THREAD

**Issue by jkbonfield** [enasequence/cramtools#25]
The CRAM file produced by htslib's tests from c1#pad1.sam fails when decoding with cramtools (both 2.1 or 3.0):

```
Failed at record 8. Here is the previously read record: [s5; flags=0; aloffset=0; mateoffset=-1; mappingQuality=0; D@0+3; D@0+3; Insertion[position=0; sequence=TA] ; P@0+1; D@6+3; qscores: ??????????]
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at net.sf.cram.CramTools.invoke(CramTools.java:93)
        at net.sf.cram.CramTools.main(CramTools.java:123)
Caused by: java.lang.RuntimeException: java.io.EOFException: End of stream.
        at net.sf.cram.encoding.reader.CramRecordReader.read(CramRecordReader.java:183)
        at net.sf.cram.build.ContainerParser.getRecords(ContainerParser.java:118)
        at net.sf.cram.build.ContainerParser.getRecords(ContainerParser.java:60)
        at net.sf.cram.Cram2Bam.main(Cram2Bam.java:222)
        ... 6 more
Caused by: java.io.EOFException: End of stream.
        at net.sf.cram.io.DefaultBitInputStream.readBits(DefaultBitInputStream.java:73)
        at net.sf.cram.encoding.huffint.Helper.read(Helper.java:164)
        at net.sf.cram.encoding.huffint.CanonicalHuffmanIntegerCodec2.read(CanonicalHuffmanIntegerCodec2.java:46)
        at net.sf.cram.encoding.huffint.CanonicalHuffmanIntegerCodec2.read(CanonicalHuffmanIntegerCodec2.java:33)
        at net.sf.cram.encoding.reader.DataReaderFactory$DefaultDataReader.readData(DataReaderFactory.java:133)
        at net.sf.cram.encoding.reader.CramRecordReader.read(CramRecordReader.java:134)
        ... 9 more
```

It also fails decoding a cram file it produced itself from the same SAM input.  See the htslib/test directory for the source.

**Comment by jkbonfield** [comment]
Similarly htslib/test/c1#pad2.sam and c1#pad3.sam fail with the same error.

**Comment by vadimzalunin** [comment]
where are the files?

**Comment by vadimzalunin** [comment]
never mind, found them on your ftp

**Comment by vadimzalunin** [comment]
fix 4a53f15a29e4c4772acc445653c237a778b03e4c
